### PR TITLE
Bump version of golang.org/x/crypto to 0.18.0 to address authentication bypass vulnerability.

### PR DESCRIPTION
**Pull request by sam-caldwell**
Bump version of golang.org/x/crypto to 0.18.0 to address authentication bypass vulnerability.

Ticket: https://github.com/Code-Hex/vz/issues/152

This PR bumps the version of a vulnerable package.

Which issue(s) this PR fixes:
Fixes #https://github.com/Code-Hex/vz/issues/152

Additional documentation
Defect identified by snyk.io scanner

### CODE REVIEW

**Comment by Code-Hex** [comment]
@sam-caldwell Could you fix compile errors?

**Comment by sam-caldwell** [comment]
Will investigate in my next spare cycles.

**Comment by cfergeau** [comment]
> Will investigate in my next spare cycles.

I've fixed this in https://github.com/cfergeau/vz/commits/go-ssh/ in the `fix go vet`commit (ran `go mod tidy`in the root dir and in èxample/*/`
github dependabot can watch dependencies and file PR automatically, I've added the configuration file for this in [use dependabot for automatic dependency upgrades](https://github.com/cfergeau/vz/commit/b842e51461ddce25ed298cb84c69dd4bbf87f28a)
I can file a PR for this if that's desirable to have (imo it is useful)

**Comment by sam-caldwell** [comment]
@cfergeau good call out with dependabot.  I use both snyk and dependabot regularly, and they are great tools for any repo.

@Code-Hex fixes should be in, just waiting for github actions to fire and run the build.  I'll have more spare cycles tomorrow morning so long as PagerDuty is willing. :-)

Cheers!

Sam

**Comment by Code-Hex** [comment]
@sam-caldwell Thank you for adding dependabot but I like renovate.

So could you revert this commit?

**Comment by sam-caldwell** [comment]
Found two additional issues in the PR.  Sorry for the messy PR...just trying to learn your codebase. :-)

- bumped the example go.mod/go.sum versions

File: .github/workflows/compile.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
   build:
     needs: formatting-check
     runs-on: ${{ matrix.os }}
-    timeout-minutes: 6
+    timeout-minutes: 15
```
Comment (Code-Hex) [inline .github/workflows/compile.yml:26]: Please Revert this one

**Review by Code-Hex — CHANGES_REQUESTED** [review]
Please only make the corrections that I specify.

File: .github/workflows/compile.yml (hunk 1 above)
Comment (sam-caldwell) [reply .github/workflows/compile.yml:26]: You asked me to help fix your broken builds.  The build is timing out.  Increasing this value now has two of the builds passing.

**Comment by Code-Hex** [comment]
@sam-caldwell I'm very sad because cannot merge this PR

**Comment by cfergeau** [comment]
@sam-caldwell I took this over in https://github.com/Code-Hex/vz/pull/155 and kept your name on the commit, let me know if this is a problem and want something different.

**Pull request closed without merging**